### PR TITLE
feat(metadata): Support directory structure

### LINKED ISSUE #42: [feature] Show directory architecture in ormb UI
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind feature

**What happened**:

Visualize the model directory structure in UI

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

### PR DESCRIPTION
**Pull request by gaocegege**
Signed-off-by: Ce Gao <gaoce.gaoc@bytedance.com>

**What this PR does / why we need it**:

Support directory structure

截屏2020-08-27 下午2 41 09

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #42 

Reference to #

**Special notes for your reviewer**:

/cc @your-reviewer

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/model/types.go [hunk 1]
```diff
@@ -11,29 +11,32 @@ type Model struct {
...
-	Training        Training          `json:"training,omitempty" yaml:"training,omitempty"`
-	Dataset         Dataset           `json:"dataset,omitempty" yaml:"dataset,omitempty"`
+	GPUType            string           `json:"gpuType,omitempty" yaml:"gpuType,omitempty"`
+	Framework          string           `json:"framework,omitempty" yaml:"framework,omitempty"`
+	Metrics            []Metric         `json:"metrics,omitempty" yaml:"metrics,omitempty"`
+	Hyperparameters    []Hyperparameter `json:"hyperparameters,omitempty" yaml:"hyperparameters,omitempty"`
+	Signature          *Signature       `json:"signature,omitempty" yaml:"signature,omitempty"`
+	Training           *Training        `json:"training,omitempty" yaml:"training,omitempty"`
+	Dataset            *Dataset         `json:"dataset,omitempty" yaml:"dataset,omitempty"`
+	DirectoryStureture []string         `json:"directoryStructure,omitempty" yaml:"directoryStructure,omitempty"`
```
Comment (simon-cj) [inline pkg/model/types.go:30]: typ error？DirectoryStureture or DirectoryStructure?

File: pkg/saver/saver.go [hunk 2]
```diff
@@ -123,4 +133,6 @@ func Tar(src string, writers ...io.Writer) error {
 
 		return nil
 	})
+
+	return structure, err
```
Comment (simon-cj) [inline pkg/saver/saver.go:137]: Suggestion：`return structure, nil`

Comment (gaocegege) [reply pkg/saver/saver.go:137]: We cannot know if the err is nil or an error.

File: pkg/model/types.go (hunk 1 above)
Comment (gaocegege) [reply pkg/model/types.go:30]: Yeah, it is a typo. good shot.

**Comment by gaocegege** [comment]
Fixed! Thanks.

PTAL again.

**Pull request merged**